checkbox-core-snap/README.md

```diff
@@ -0,0 +1,137 @@
+# Building checkbox-core-snap
+This snap is the core reusable part of checkbox, it includes all the utilities
+and some core providers.
+
+## Building guide
+This section convers how to build the snap in a normal settings. 
+via either the `multipass` or the `lxd` backend. This guide will focus on the latter
+but with small adjustments can be used for the former. Also, this guide will
+build the `series22` snap, adjust the commands to build any other.
+
+In order to build the snap you are going to need the following:
+1. **snapcraft:** Installed via `snap install snapcraft --classic` or 
+`snap install snapcraft --classic --channel=4.x` for series16
+2. **python3:** Installed via `apt` or any other packaging solution
+3. **setuptools_scm:** Installed via pip
+4. **rsync:** Installed via apt or any other packaging solution
+
+After installing all the dependencies do the following:
```

Comment:
Perhaps explain what `prepare.sh` is about. For example:

"After installing all the dependencies and cloning the repo, to output the snapcraft.yaml for `core22` as an example, do the following:"